From bc587ccd5d167a5c99fe3227d0b526024bea73fc Mon Sep 17 00:00:00 2001
From: wudihui <wudihui@xiaomi.corp-partner.google.com>
Date: Wed, 13 May 2026 16:51:10 +0800
Subject: [PATCH] ANDROID: GKI: update symbol list file for xiaomi

7 function symbol(s) added
  'int __traceiter_android_vh_cma_alloc_lat_end(void*, unsigned long long, unsigned long)'
  'int __traceiter_android_vh_cma_alloc_lat_start(void*, unsigned long long*)'
  'int __traceiter_android_vh_dma_heap_buffer_alloc_lat_end(void*, unsigned long long, size_t, struct dma_buf*)'
  'int __traceiter_android_vh_dma_heap_buffer_alloc_lat_start(void*, unsigned long long*)'
  'int __traceiter_android_vh_mm_direct_reclaim_end(void*, unsigned int, u64)'
  'int __traceiter_android_vh_mm_direct_reclaim_start(void*, u64*)'
  'unsigned long try_to_free_pages(struct zonelist*, int, gfp_t, nodemask_t*)'

6 variable symbol(s) added
  'struct tracepoint __tracepoint_android_vh_cma_alloc_lat_end'
  'struct tracepoint __tracepoint_android_vh_cma_alloc_lat_start'
  'struct tracepoint __tracepoint_android_vh_dma_heap_buffer_alloc_lat_end'
  'struct tracepoint __tracepoint_android_vh_dma_heap_buffer_alloc_lat_start'
  'struct tracepoint __tracepoint_android_vh_mm_direct_reclaim_end'
  'struct tracepoint __tracepoint_android_vh_mm_direct_reclaim_start'

Bug: 514559198
Bug: 506924601

Change-Id: I5f6df0891c70ccfa9f9feb99de27d208cf5f5246
Signed-off-by: wudihui <wudihui@xiaomi.corp-partner.google.com>
(cherry picked from commit bb3ee689d8378a4c1f5d320324d474204cae288a)
---
 gki/aarch64/abi.stg        | 166 +++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/xiaomi |  13 +++
 2 files changed, 179 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index ca87c01b49ce..cc5a169092a0 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -393290,6 +393290,14 @@ function {
   parameter_id: 0x1b52b59a
   parameter_id: 0x08798138
 }
+function {
+  id: 0x37ade47b
+  return_type_id: 0x33756485
+  parameter_id: 0x1e85f467
+  parameter_id: 0x6720d32f
+  parameter_id: 0xf1a6dfed
+  parameter_id: 0x2fb2b385
+}
 function {
   id: 0x37d5564f
   return_type_id: 0x33756485
@@ -412037,6 +412045,14 @@ function {
   parameter_id: 0x791165e9
   parameter_id: 0xc9082b19
 }
+function {
+  id: 0x98c5c985
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0xd5df6730
+  parameter_id: 0xf435685e
+  parameter_id: 0x00c83ba6
+}
 function {
   id: 0x98c5cae2
   return_type_id: 0x6720d32f
@@ -412278,6 +412294,13 @@ function {
   parameter_id: 0x165dc292
   parameter_id: 0x39947122
 }
+function {
+  id: 0x98f41aa6
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0xd5df6730
+  parameter_id: 0x33756485
+}
 function {
   id: 0x98fb76e9
   return_type_id: 0x6720d32f
@@ -414886,6 +414909,13 @@ function {
   parameter_id: 0x1a1051a0
   parameter_id: 0x6720d32f
 }
+function {
+  id: 0x9a9d2737
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x4585663f
+  parameter_id: 0x92233392
+}
 function {
   id: 0x9aa0b156
   return_type_id: 0x6720d32f
@@ -417010,6 +417040,12 @@ function {
   parameter_id: 0x3e6239e1
   parameter_id: 0x18bd6530
 }
+function {
+  id: 0x9b5025e2
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x3fe7e06b
+}
 function {
   id: 0x9b506e5c
   return_type_id: 0x6720d32f
@@ -442361,6 +442397,24 @@ elf_symbol {
   type_id: 0x9bef35b0
   full_name: "__traceiter_android_vh_cma_alloc_finish"
 }
+elf_symbol {
+  id: 0xd61f4b99
+  name: "__traceiter_android_vh_cma_alloc_lat_end"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x906d1a62
+  type_id: 0x98f41aa6
+  full_name: "__traceiter_android_vh_cma_alloc_lat_end"
+}
+elf_symbol {
+  id: 0x55cd1855
+  name: "__traceiter_android_vh_cma_alloc_lat_start"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x6a58317c
+  type_id: 0x9b5025e2
+  full_name: "__traceiter_android_vh_cma_alloc_lat_start"
+}
 elf_symbol {
   id: 0x9cbc7813
   name: "__traceiter_android_vh_cma_alloc_retry"
@@ -442667,6 +442721,24 @@ elf_symbol {
   type_id: 0x9b6af5ed
   full_name: "__traceiter_android_vh_dma_heap_buffer_alloc_end"
 }
+elf_symbol {
+  id: 0xdd287008
+  name: "__traceiter_android_vh_dma_heap_buffer_alloc_lat_end"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x3398120a
+  type_id: 0x98c5c985
+  full_name: "__traceiter_android_vh_dma_heap_buffer_alloc_lat_end"
+}
+elf_symbol {
+  id: 0x345346b5
+  name: "__traceiter_android_vh_dma_heap_buffer_alloc_lat_start"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x6a58317c
+  type_id: 0x9b5025e2
+  full_name: "__traceiter_android_vh_dma_heap_buffer_alloc_lat_start"
+}
 elf_symbol {
   id: 0xbcd1a27c
   name: "__traceiter_android_vh_dma_heap_buffer_alloc_start"
@@ -444494,6 +444566,15 @@ elf_symbol {
   type_id: 0x9b82a930
   full_name: "__traceiter_android_vh_mm_customize_zone_pageset"
 }
+elf_symbol {
+  id: 0x67b594b4
+  name: "__traceiter_android_vh_mm_direct_reclaim_end"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x49b4cb37
+  type_id: 0x9a9d2737
+  full_name: "__traceiter_android_vh_mm_direct_reclaim_end"
+}
 elf_symbol {
   id: 0x9f58159a
   name: "__traceiter_android_vh_mm_direct_reclaim_enter"
@@ -444512,6 +444593,15 @@ elf_symbol {
   type_id: 0x9b7ba7c5
   full_name: "__traceiter_android_vh_mm_direct_reclaim_exit"
 }
+elf_symbol {
+  id: 0x870be3b8
+  name: "__traceiter_android_vh_mm_direct_reclaim_start"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x67fdab9a
+  type_id: 0x9b17d9b6
+  full_name: "__traceiter_android_vh_mm_direct_reclaim_start"
+}
 elf_symbol {
   id: 0x4df97812
   name: "__traceiter_android_vh_mm_do_madvise_bypass"
@@ -450326,6 +450416,24 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_cma_alloc_finish"
 }
+elf_symbol {
+  id: 0xdbbfd393
+  name: "__tracepoint_android_vh_cma_alloc_lat_end"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_cma_alloc_lat_end"
+}
+elf_symbol {
+  id: 0x1c09c9eb
+  name: "__tracepoint_android_vh_cma_alloc_lat_start"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_cma_alloc_lat_start"
+}
 elf_symbol {
   id: 0xaf598d05
   name: "__tracepoint_android_vh_cma_alloc_retry"
@@ -450632,6 +450740,24 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_dma_heap_buffer_alloc_end"
 }
+elf_symbol {
+  id: 0x30beda82
+  name: "__tracepoint_android_vh_dma_heap_buffer_alloc_lat_end"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_dma_heap_buffer_alloc_lat_end"
+}
+elf_symbol {
+  id: 0x19f49f43
+  name: "__tracepoint_android_vh_dma_heap_buffer_alloc_lat_start"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_dma_heap_buffer_alloc_lat_start"
+}
 elf_symbol {
   id: 0x9d32ef5e
   name: "__tracepoint_android_vh_dma_heap_buffer_alloc_start"
@@ -452459,6 +452585,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_mm_customize_zone_pageset"
 }
+elf_symbol {
+  id: 0xf4aa1b92
+  name: "__tracepoint_android_vh_mm_direct_reclaim_end"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_mm_direct_reclaim_end"
+}
 elf_symbol {
   id: 0xd333a65c
   name: "__tracepoint_android_vh_mm_direct_reclaim_enter"
@@ -452477,6 +452612,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_mm_direct_reclaim_exit"
 }
+elf_symbol {
+  id: 0xe95ea39a
+  name: "__tracepoint_android_vh_mm_direct_reclaim_start"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_mm_direct_reclaim_start"
+}
 elf_symbol {
   id: 0x56c6a2a4
   name: "__tracepoint_android_vh_mm_do_madvise_bypass"
@@ -516309,6 +516453,15 @@ elf_symbol {
   type_id: 0x360a883f
   full_name: "try_to_free_mem_cgroup_pages"
 }
+elf_symbol {
+  id: 0xb95bc64f
+  name: "try_to_free_pages"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xb1b855e3
+  type_id: 0x37ade47b
+  full_name: "try_to_free_pages"
+}
 elf_symbol {
   id: 0x8fa1e478
   name: "try_to_writeback_inodes_sb"
@@ -527956,6 +528109,8 @@ interface {
   symbol_id: 0xe0295cb9
   symbol_id: 0xa9ea0208
   symbol_id: 0x606360ad
+  symbol_id: 0xd61f4b99
+  symbol_id: 0x55cd1855
   symbol_id: 0x9cbc7813
   symbol_id: 0x1104ae4c
   symbol_id: 0x943ca30b
@@ -527990,6 +528145,8 @@ interface {
   symbol_id: 0x5f33b294
   symbol_id: 0x6664371a
   symbol_id: 0x94b48bb0
+  symbol_id: 0xdd287008
+  symbol_id: 0x345346b5
   symbol_id: 0xbcd1a27c
   symbol_id: 0x5e9ed424
   symbol_id: 0xb7d9e03a
@@ -528193,8 +528350,10 @@ interface {
   symbol_id: 0xcbb71c7c
   symbol_id: 0x2be8352c
   symbol_id: 0xa792d107
+  symbol_id: 0x67b594b4
   symbol_id: 0x9f58159a
   symbol_id: 0x29c67d40
+  symbol_id: 0x870be3b8
   symbol_id: 0x4df97812
   symbol_id: 0x3fe16974
   symbol_id: 0x7f7c30a9
@@ -528841,6 +529000,8 @@ interface {
   symbol_id: 0x74d9da97
   symbol_id: 0xdb2e754e
   symbol_id: 0x02efcc97
+  symbol_id: 0xdbbfd393
+  symbol_id: 0x1c09c9eb
   symbol_id: 0xaf598d05
   symbol_id: 0x62c56ae6
   symbol_id: 0xb122c55d
@@ -528875,6 +529036,8 @@ interface {
   symbol_id: 0xe829284e
   symbol_id: 0x7bbd7744
   symbol_id: 0x30442e2e
+  symbol_id: 0x30beda82
+  symbol_id: 0x19f49f43
   symbol_id: 0x9d32ef5e
   symbol_id: 0x325e746a
   symbol_id: 0x066e59a8
@@ -529078,8 +529241,10 @@ interface {
   symbol_id: 0x329b593e
   symbol_id: 0x4c4316c6
   symbol_id: 0xc25513c9
+  symbol_id: 0xf4aa1b92
   symbol_id: 0xd333a65c
   symbol_id: 0xddcff44a
+  symbol_id: 0xe95ea39a
   symbol_id: 0x56c6a2a4
   symbol_id: 0x533ca98e
   symbol_id: 0x046ed8db
@@ -536168,6 +536333,7 @@ interface {
   symbol_id: 0x2f855953
   symbol_id: 0x68605044
   symbol_id: 0x8814f5b8
+  symbol_id: 0xb95bc64f
   symbol_id: 0x8fa1e478
   symbol_id: 0xe54aec5b
   symbol_id: 0xce4acc7b
diff --git a/gki/aarch64/symbols/xiaomi b/gki/aarch64/symbols/xiaomi
index ac32c6fce67e..0bd1573d91a7 100644
--- a/gki/aarch64/symbols/xiaomi
+++ b/gki/aarch64/symbols/xiaomi
@@ -644,4 +644,17 @@ __tracepoint_android_vh_cma_release_bypass
 __traceiter_android_vh_cma_release_bypass
 __tracepoint_android_vh_cma_alloc_end
 __traceiter_android_vh_cma_alloc_end
+__traceiter_android_vh_cma_alloc_lat_start
+__tracepoint_android_vh_cma_alloc_lat_start
+__traceiter_android_vh_cma_alloc_lat_end
+__tracepoint_android_vh_cma_alloc_lat_end
+__traceiter_android_vh_dma_heap_buffer_alloc_lat_start
+__tracepoint_android_vh_dma_heap_buffer_alloc_lat_start
+__traceiter_android_vh_dma_heap_buffer_alloc_lat_end
+__tracepoint_android_vh_dma_heap_buffer_alloc_lat_end
+__tracepoint_android_vh_mm_direct_reclaim_start
+__traceiter_android_vh_mm_direct_reclaim_start
+__tracepoint_android_vh_mm_direct_reclaim_end
+__traceiter_android_vh_mm_direct_reclaim_end
+try_to_free_pages
 
-- 
2.55.0

